kebechet/managers/thoth_advise/thoth_advise.py

```diff
@@ -0,0 +1,149 @@
+        self.repo.index.add(files)
+        self.repo.index.commit(commit_msg)
+        self.repo.remote().push(branch_name, force=force_push)
+
+    def _open_merge_request(
+        self, branch_name: str, labels: list, files: list
+    ) -> typing.Optional[int]:
+        """Open a pull/merge request for dependency update."""
+        commit_msg = "Auto generated update"
+        body = "Pipfile.lock updated by kebechet-thoth manager"
+
+        # Delete branch if it didn't change Pipfile.lock
+        diff = self.repo.git.diff("master", files)
+        if diff == "":
+            return
+
+        # push force always to keep branch up2date with the recent master and avoid merge conflicts.
+        LOGGER_.info('Pushing changes')
+        self._git_push(":pushpin: " + commit_msg, branch_name, files, force_push=True)
+
+        # Check if the merge request already exists
+        for mr in self._cached_merge_requests:
+            if mr.head_branch_name == branch_name:
+                LOGGER_.info('Merge request already exists, updating...')
+                return
+
+        LOGGER_.info('Opening merge request')
+        merge_request = self.sm.open_merge_request(
+            commit_msg, branch_name, body, labels
+        )
+        return merge_request
+
+    @staticmethod
+    def _write_advise(adv_results: list):
+        lock_info = adv_results[0]["report"][0][1]["requirements_locked"]
+        with open("Pipfile.lock", "w+") as f:
+            LOGGER_.info('Writing to Pipfile.lock')
+            LOGGER_.debug(f"{json.dumps(lock_info)}")
+            f.write(json.dumps(lock_info))
+        return
+
+    def _issue_advise_error(self, adv_results: list, labels: list):
+        """Create an issue if advise fails."""
+        error_info = adv_results[0]["report"][0][0][0]
+        justification = error_info["justification"]
+        type_ = error_info["type"]
+        LOGGER_.warning('Error type: {type_}')
+        checksum = hashlib.md5(justification.encode("utf-8")).hexdigest()[:10]
+        LOGGER_.info('Creating issue')
+        self.sm.open_issue_if_not_exist(
+            f"{checksum}-{type_}: Automated kebechet thoth-advise Issue",
+            lambda: justification,
+            labels=labels,
+        )
+
+    def run(self, labels: list):
+        """Run Thoth Advising Bot."""
+        with cloned_repo(self.service_url, self.slug, depth=1) as repo:
+            self.repo = repo
+            branch_name = self._construct_branch_name()
+            branch = self.repo.git.checkout("-B", branch_name)
+            self._cached_merge_requests = self.sm.repository.merge_requests
+            if os.path.isfile("Pipfile"):
+                res = lib.advise_here()
+                for i in range(1, 11):
+                    if res is not None:
+                        break
+                    LOGGER_.info(f"Advising failed, retrying ({i}/10)")
+                    res = lib.advise_here(force=True)
```

Comment:
Without `force` set to True. If requirements did not changed, we don't want to recompute results.

Reply:
The force is only if result is `None`, if a submission was made my first call does not force so it should get that result if it exists in the cache

Reply:
I'm not sure if I follow here. Caching is done on the server side (backend should know what can be cached, for how long and how), the client should not be aware of it and dummy in this way.

Reply:
If the first advise fails it caches the failure so you have to force

Reply:
Yes, but that's caused by the recent changes (bug). This is not something that should be happening and kebechet should handle.